python/src/opendp/extras/aim_new.py

```diff
@@ -0,0 +1,205 @@
+        t = 0                                                   # Algo 2 line 6
+        
+        # important arrays updated outside of the loop
+        sigmas = [math.sqrt(T / (2 * args.alpha * args.rho))]   # Algo 2 line 4
+        epsilons = [0] # I initialize this with 0 bc there is no epsilon 0 and want to keep indices consistent with sigmas
+        all_marginals = [] # M_r(D) for all r in W_plus; this is a list of lists; NEED_TO_INITIALIZE
+        selected_queries = [] # r_1 through r_t-1
+        selected_query_indices = [] # indices of queries in selected_queries
+        selected_noisy_marginals = [] # this is an array of arrays y_t tilde
+
+        # updated synthetic datasets
+        p_hat = np.zeros((n, d)) # the last generated synthetic dataset
+        p_hat_new = np.zeros((n, d)) # the new synthetic dataset after each measurement
+            
+        # all_marginals initialization - changed to transformation
+        marginal_transformations = [make_distribution_marginal_transformation(query) for query in W_plus]
+        for marginal_fn in marginal_transformations:
+            marginal = marginal_fn(data)
+            all_marginals.append(marginal)
+
+        # Algo 3: initialize p_t
+        size_one_queries, size_one_indices = zip(*[
+            (query, index) for index, query in enumerate(W_plus) if len(query) == 1                                 # Algo 3 line 1
+        ])
+        size_one_measurements = []
+        for query, index in zip(size_one_queries, size_one_indices):
+            t += 1                                                                                                  # Algo 3 line 2
+            sigmas.append(sigmas[0])                                                                                # Algo 3 line 2
+            gaussian = dp.m.make_gaussian(input_domain = input_domain, input_metric = input_metric, scale = sigmas[t])
+            y_t = np.array([gaussian(val) for val in all_marginals[index]])                                         # Algo 3 line 3 
+            size_one_measurements.append(y_t)                                                                       # Algo 3 line 3; this array should be reset every time we start measuring again
+            rho_used += 1/(2 * sigmas[t]**2)                                                                        # Algo 3 line 4
+        p_hat = np.zeros((n, d))                                                                                    # Algo 3 line 6 -- FIX_THIS; need to add the import from Private-PGM
+
+        # initialize sigmas and epsilons
+        sigmas.append(sigmas[0])
+        epsilons.append(math.sqrt(8 * (1 - args.alpha) * args.rho / args.T)) # Algo 2 line 9
+
+        # large while loop, Algo 2 line 10
+        while rho_used < args.rho:
+            # important arrays within the loop
+            C_t = []
+            C_t_indices = [] # indices in W_plus of each query in C_t
```

Comment:
While there is an argument for reusing the same variable names as an algorithm in a paper, it is generally preferable to use informative english names. Implementations tend to stray from the paper, either due to substantive adjustments to what was written in the paper, or because the code takes more lines to implement. Ties back to specific variables in a paper can be made via comments instead, if not clear.